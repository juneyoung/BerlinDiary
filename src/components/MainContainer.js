import React, { Component } from 'react';
import MainBanner from './MainBanner';
import LinearContentsContainer from './LinearContentsContainer'
import ModalContent from './modal/ModalContent'
import Dropbox from 'dropbox';

export default class MainContainer extends Component {
	constructor (props) {
		super(props);

		let sampleData = {

			20180630 : [
				{
					seq : 0
					, date : 20180330063000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/media/morning.mp4'
					, title : 'Morning Impression of Berlin'
					, desc : 'Good morning, Berlin!'
				}
				, {
					seq : 1
					, date : 20180330123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
				, {
					seq : 2
					, date : 20180330123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
				, {
					seq : 3
					, date : 20180330123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
				, {
					seq : 4
					, date : 20180330123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
				, {
					seq : 5
					, date : 20180330123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
			], 
			20180629 : [
				{
					seq : 0
					, date : 20180329063000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/media/morning.mp4'
					, title : 'Morning Impression of Berlin'
					, desc : 'Good morning, Berlin!'
				}
				, {
					seq : 1
					, date : 20180329123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
			],
			20180628 : [
				{
					seq : 0
					, date : 20180328063000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/media/morning.mp4'
					, title : 'Morning Impression of Berlin'
					, desc : 'Good morning, Berlin!'
				}
				, {
					seq : 1
					, date : 20180328123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
			],
			20180627 : [
				{
					seq : 0
					, date : 20180327063000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/media/morning.mp4'
					, title : 'Morning Impression of Berlin'
					, desc : 'Good morning, Berlin!'
				}
				, {
					seq : 1
					, date : 20180327123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
			],
			20180626 : [
				{
					seq : 0
					, date : 20180326063000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/media/morning.mp4'
					, title : 'Morning Impression of Berlin'
					, desc : 'Good morning, Berlin!'
				}
				, {
					seq : 1
					, date : 20180325123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/afternoon.png'
					, title : 'Lunch in Berlin, second day'
					, desc : 'Blah Blah Blah Blah Blah Blah Blah'
				}
			],

			20180625 : [
				{
					seq : 0
					, date : 20180325083000
					, type : 'video'
					, mediaUrl : 'http://127.0.0.1:3000/assets/media/dock.mp4'
					, title : 'Arrived in Berlin International airport'
					, desc : 'Arrived in Berlin. Begin life as Berliner'
				}
				, {
					seq : 1
					, date : 20180325123000
					, type : 'photo'
					, mediaUrl : 'http://127.0.0.1:3000/assets/images/beer.png'
					, title : 'Lunch in Berlin, first day'
					, desc : 'Very first lunch in Germany in 2018'
				}
			]
		};


		let dropbox = null;
		try {
			let dropBoxInfo = this.props.apiKeys.filter(mem => {
				return mem.vendor === 'dropbox';
			})
			console.log('dropBoxInfo :: ', dropBoxInfo);
			dropbox = new Dropbox.Dropbox({ accessToken:  dropBoxInfo[0].accessToken });
		} catch(dropboxException) {
			console.error('An Error Occurs while initialize Dropbox. set dropbox variable as null :: ', dropboxException);
		}
		
		this.state = {
			debuggingName : 'Main Container Area'
			, sampleData : sampleData
			, dropbox : dropbox
		}
	}

	/*
		// ajax recursive 는 어떻게 처리해야 할까...
		listRecursiveDropboxFiles = (path, func) => {
			path = path || '';
			console.log('listing dropboxFiles path :: ', path);
			this.state.dropbox.filesListFolder({path: path})
			  .then(function(response) {
			    console.log('data fetched from dropbox successfully :: ', response.entries);
			    var entries = response.entries || [];
			    for(idx in entries) {
			    	let entry = entries[idx];
			    	if(entry['.tag'] === 'folder') {
			    		path_lower: "/0625"
			    	}
			    }
			  })
			  .catch(function(error) {
			    console.log(error);
			});
		}
	*/

	regroup = (coll, f) => {
		return coll.reduce(function(acc, x) {
			let k = f(x);
			acc[k] = (acc[k] || []).concat(x);
			return acc;
		}, {});
	}


	componentDidMount = () => {
		let currentThis = this;
		if(!! this.state.dropbox) {
			console.log('on ComponentDidMount !!');
			this.state.dropbox.filesListFolder({path: '', recursive: true})
			  .then(function(response) {
			    console.log('data fetched from dropbox successfully :: ', response.entries);
			    let entries = response.entries || [];
			    let distinguished = currentThis.regroup(entries, (x) => { return x['.tag']; });
			    console.log('distinguished Entries :: ', distinguished);
			    window.d = distinguished;

			    console.log('#0. distinguished folder :: ', distinguished.folder)
			    let folders = distinguished.folder;
			    let files = distinguished.file;
			    console.log('#0. distinguished file :: ', files)
			    let tempData = currentThis.regroup(folders, (x) => { return x['name'] });
			    console.log('#1.  tempData :: ', tempData);
			    for (let date of Object.keys(tempData)) {
			    	console.log('#2.  tempData-date :: ', date);
			    	let dateElement = tempData[date][0] || {};
			    	files.forEach(memFile => {
			    		console.log('#3. memFile :: ', memFile);
			    		console.log('#4. tempData :: ', tempData);
			    		if(memFile['path_lower'].indexOf(date) > -1) {
			    			let testTemp = Object.assign({}, memFile);
			    			/*
								엘리먼트 맞춰줘야 데이터 제대로 나오겠다...
			    			*/
			    			dateElement['items'] = dateElement['items'] || [];
			    			dateElement['items'] = dateElement['items'].concat(memFile);
			    		}
			    	});
			    }
			    console.log('regrouped Data Array :: ', JSON.stringify(tempData));
			    // currentThis.setState({ sampleData :  tempData });
			  })
			  .catch(function(error) {
			    console.log(error);
			});	
		} else {
			console.log('Dropbox data is null');
		}
	}


	render () {
		return (
			<div className='mainContainer'>
				{/* <ModalContent/> */}
				<MainBanner />
				<div className='contentsHolder'>
					<div className='contentBg dimmed'></div>
					<LinearContentsContainer posts={this.state.sampleData}/>
				</div>
			</div>
		)
	}
}