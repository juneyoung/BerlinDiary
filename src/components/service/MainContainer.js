import React, { Component } from 'react';
import MainBanner from './MainBanner';
import LinearContentsContainer from './LinearContentsContainer'
// import ModalContent from './modal/ModalContent'
import sampleData from '../../assets/data/testData.json';
import ElasticsearchClient from '../../assets/scripts/ElasticsearchClient';

// 여기서 던지는 게 맞을 거 같은데...
import Modal from 'react-modal'
import PostView from './PostView'

const postModalCustomStyles = {
  content : {
    top                   : '50%',
    left                  : '50%',
    right                 : 'auto',
    bottom                : 'auto',
    marginRight           : '-50%',
    transform             : 'translate(-50%, -50%)',
    width				  : '80%'	
  }
};


export default class MainContainer extends Component {
	constructor (props) {
		super(props);
		this.state = {
			debuggingName : 'Main Container Area'
			, data : sampleData
			, modalIsOpen : false
			, modalContent : {}
		}
		this.openModal = this.openModal.bind(this);
    	this.afterOpenModal = this.afterOpenModal.bind(this);
    	this.closeModal = this.closeModal.bind(this);
	}

	regroup = (coll, f) => {
		return coll.reduce(function(acc, x) {
			let k = f(x);
			acc[k] = (acc[k] || []).concat(x);
			return acc;
		}, {});
	}

	listingPosts = () => {
		console.log('listingPosts');
		let ref = this;
		console.log('component did mount called');
		ElasticsearchClient.search({
			index : 'berlin'
			, type: 'post'
			, q: '*:*'
			, sort : "_id:desc"	// : and comma string
		}, function(err, resp) {
			if(err) {
				alert('An Error occurs while fetching data from Elasticsearch!');
				console.error('An error occurs while listing post', err);
			} else {

				function yyyymmdd(date){
					if( typeof date === 'string') {
						return date.substring(0, date.indexOf('T')).replace(/-/g, '');
					}
					var mm = date.getMonth() + 1; // getMonth() is zero-based
					var dd = date.getDate();
					return [ date.getFullYear(), (mm>9 ? '' : '0') + mm, (dd>9 ? '' : '0') + dd ].join('');
				}

				let dataList = resp.hits.hits.map((mem, i) => {
					console.log('_data_id ', mem._id);
					let tempObj = {};
					tempObj.seq = mem._id;
					tempObj.type = 'photo';
					tempObj.mediaUrl = 'http://127.0.0.1:3000/assets/images/afternoon.png';
					tempObj.title = mem._source.title || '';
					tempObj.desc = mem._source.description || '';
					tempObj.date = (!!mem._source.created) ? yyyymmdd(mem._source.created) : yyyymmdd(new Date()); 
					tempObj.created = mem._source.created || '';
					tempObj.changed = mem._source.changed || '';
					return tempObj;
				});

				dataList = ref.regroup(dataList, function(x){ return x.date; })
				// console.log('fetched Data :: ', dataList);
				ref.setState({ data : dataList });
			}
		})
	}

	openModal(data) {
		this.setState({ modalIsOpen: true, modalContent: (data || {}) });
	}

	afterOpenModal() {
		console.log('define something after open modal');
	}

	closeModal() {
		this.setState({modalIsOpen: false, modalContent: {}});
	}

	componentDidMount = () => {
		this.listingPosts();
	}

	componentWillMount () {
		Modal.setAppElement('body');
	}

	render () {
		return (
			<div className='mainContainer'>
				<MainBanner />
				<div className='contentsHolder'>
					<div className='contentBg dimmed'></div>
					<LinearContentsContainer posts={this.state.data} onPostSelect={ this.openModal }/>
				</div>

				<Modal
					isOpen={ this.state.modalIsOpen }
					onAfterOpen={ this.afterOpenModal }
					onRequestClose={ this.closeModal }
					style={ postModalCustomStyles }
					contentLabel="Post Modal"
				>
					<h2 ref={subtitle => this.subtitle = subtitle}>{ this.state.modalContent.title || 'Title is empty' }</h2>
					<PostView data={ this.state.modalContent } closeEvt={ this.closeModal } />
				</Modal>
			</div>
		)
	}
}