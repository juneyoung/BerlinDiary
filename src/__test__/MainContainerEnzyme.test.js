import React from 'react';
import { shallow } from 'enzyme';
import MainContainer from '../components/MainContainer';

/*
	describe
	it
	expect
*/

describe('MainContainer jest', () => {
	let component = null;

	it("MainContainer render validation", () => {
		// enzyme 은 쿼테이션으로 싸면 오류남 > Invariant Violation: ReactShallowRenderer render(): Invalid component element
		// component = shallow('<MainContainer/>');
		component = shallow(<MainContainer/>);
	});

	// it('Checking Initial snapshot', () => {
	// 	const tree = component.toJSON();
	// 	expect(tree).toMatchSnapshot();
	// });

	it('Matching to snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});