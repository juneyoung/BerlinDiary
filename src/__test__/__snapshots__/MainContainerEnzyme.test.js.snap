// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer jest Matching to snapshot 1`] = `
<div
  className="mainContainer"
>
  <MainBanner />
  <div
    className="contentsHolder"
  >
    1. put Title inside here 
    <br />
    2. put PullToRefresh List...
  </div>
</div>
`;
